### ISSUE TITLE
`NullReferenceException` in `CollectionAnalyzer.ShouldAnalyzeVariableType`

### ISSUE THREAD

**Issue by melotic** [fluentassertions/fluentassertions.analyzers#215]
Hello, we've run in this issue below, which began in v0.22.0. (https://github.com/microsoft/component-detection/pull/726)

Interestingly, this only happens when our CodeQL CI attempts to auto build our project, but not during a regular `dotnet build`. Building locally and in our CI works just fine.

Example build: https://github.com/microsoft/component-detection/actions/runs/5905756165/job/16020455125#step:4:230

```
Failed to analyze expression in FluentAssertions.Analyzers.CollectionShouldBeEmptyAnalyzer. [/home/runner/work/component-detection/component-detection/test/Microsoft.ComponentDetection.Orchestrator.Tests/Microsoft.ComponentDetection.Orchestrator.Tests.csproj]
System.NullReferenceException: Object reference not set to an instance of an object. [/home/runner/work/component-detection/component-detection/test/Microsoft.ComponentDetection.Orchestrator.Tests/Microsoft.ComponentDetection.Orchestrator.Tests.csproj]
   at FluentAssertions.Analyzers.CollectionAnalyzer.ShouldAnalyzeVariableType(ITypeSymbol type, SemanticModel semanticModel) [/home/runner/work/component-detection/component-detection/test/Microsoft.ComponentDetection.Orchestrator.Tests/Microsoft.ComponentDetection.Orchestrator.Tests.csproj]
   at FluentAssertions.Analyzers.FluentAssertionsAnalyzer`1.ShouldAnalyzeVariableTypeCore(ITypeSymbol type, SemanticModel semanticModel) [/home/runner/work/component-detection/component-detection/test/Microsoft.ComponentDetection.Orchestrator.Tests/Microsoft.ComponentDetection.Orchestrator.Tests.csproj]
   at FluentAssertions.Analyzers.FluentAssertionsAnalyzer`1.AnalyzeExpression(ExpressionSyntax expression, SemanticModel semanticModel) [/home/runner/work/component-detection/component-detection/test/Microsoft.ComponentDetection.Orchestrator.Tests/Microsoft.ComponentDetection.Orchestrator.Tests.csproj]
```

**Comment by Meir017** [comment]
@melotic can you try version `0.22.1` or `0.23.1`? there should be more logs to troubleshoot this https://github.com/microsoft/component-detection/pull/726

**Comment by Meir017** [comment]
resovlved by #217

**Comment by melotic** [comment]
> resolved by #217 @melotic can you please test with the new version `0.24.0`

Yep, this fixed it! Thanks for your help & quick turn around
